features/step_definitions/git-flow.js

```diff
@@ -104,90 +16,44 @@ function promiseToWait(timeToWait) {
-
-        if (buildData.length === MAX_PAGE_COUNT) {
-            return searchForBuild(screwdriverInstance, desiredSha, pageCounter + 1)
-            .then((nextPage) => result.concat(nextPage));
-        }
-
-        return result;
-    });
-}
-
-/**
- * Perisistently ping the API until the build data is available
- * @method waitForBuild
- * @param  {String}     screwdriverInstance  Specific Screwdriver instance to query against
- * @param  {String}       desiredSha         The SHA that the build is using
- * @return {Promise}
- */
-function waitForBuild(screwdriverInstance, desiredSha) {
-    console.log('    (Waiting for build to exist....)');
-
-    return searchForBuild(screwdriverInstance, desiredSha)
-    .then((buildData) => {
-        if (buildData.length !== 0) {
-            return buildData;
-        }
-
-        return promiseToWait(3)
-            .then(() => searchForBuild(screwdriverInstance, desiredSha));
-    });
-}
-
 module.exports = function server() {
     // eslint-disable-next-line new-cap
-    this.Before(() => {
+    this.Before({
+        tags: ['@gitflow'],
+        timeout: 60000
+    }, () => {
         this.instance = 'https://api.screwdriver.cd';
-        this.repositoryOwner = 'screwdriver-cd';
-        this.repository = 'garbage-repository-ignore-this';
-        this.testBranch = 'testBranch';
-        const params = {
-            user: this.repositoryOwner,
-            repo: this.repository,
-            ref: `heads/${this.testBranch}`
-        };
+        this.branch = 'darrenBranch';
+        this.repoOrg = 'screwdriver-cd-test';
+        this.repoName = 'functional-git';
+        this.pipelineId = '2e0138dfa7c4ff83720dc0cd510d2252a3398fc3';  // TODO: determine dynamically
 
-        // PR creation requires authentication
-        github.authenticate({
-            type: 'oauth',
-            token: this.github_token
-        });
+        // Reset shared information
+        this.pullRequestNumber = null;
 
-        return github.gitdata.getReference(params)
-            // If branch exists, delete it
-            .then(() => github.gitdata.deleteReference(params), () => {});
+        return request({  // TODO : perform this in the before-hook for all func tests
+            method: 'GET',
+            url: `${this.instance}/v4/auth/token?access_key=${this.accessKey}`,
+            followAllRedirects: true,
+            json: true
+        }).then((response) => {
+            this.jwt = response.body.token;
+        }).then(() =>
+            github.cleanUpRepository(this.gitToken, this.branch, this.repoOrg,
+                this.repoName)
+        );
     });
 
-    this.Given(/^an existing pipeline$/, () => {
+    this.Given(/^an existing pipeline$/, () =>
         request({
             uri: `${this.instance}/v4/pipelines`,
             method: 'POST',
             auth: {
-                username: this.username,
+                user: this.username,
```

Comment:
`username` might be used later in the authorization functional test, but I can remove it here and add it in a later pull request if it's needed.